src/coreComponents/common/format/table/TableFormatter.cpp

```diff
@@ -96,315 +134,525 @@ string buildCell( TableLayout::Alignment const alignment, string_view value, int
+            }
+            updateCellMaxLength( cellsDataLayout[rowIdx2][idxColumn], maxColumnSize, previousDataCellTemp );
+            if( cellsDataLayout[rowIdx2][idxColumn].m_cellType == CellType::Separator )
+            {
+              size_t const additionnalMargin = (idxColumn == numColumns - 1) ?
+                                               tableLayout.getBorderMargin() * 2 + 2 :
+                                               tableLayout.getColumnMargin();
+              cellsDataLayout[rowIdx2][idxColumn].m_lines[0]  = std::string( maxColumnSize + additionnalMargin, '-' );
+            }
+          }
+        }
+
+        accMaxStringColumn[rowIdx] = 0;
+      }
+      else
+      {
+        size_t const cellDataLength = getMaxLineLength( cellsDataLayout[rowIdx][idxColumn].m_lines );
+        maxColumnSize = std::max( {maxColumnSize, cellDataLength} );
+      }
 
+      updateCellMaxLength( dataCell, maxColumnSize, previousDataCell );
 
-void TableTextFormatter::buildTableSeparators( std::vector< TableLayout::Column > const & columns,
-                                               string & topSeparator,
-                                               string & sectionSeparatingLine ) const
-{
-  { // section separator line
-    integer const columnMargin = m_tableLayout.getColumnMargin();
-    integer const borderMargin = m_tableLayout.getBorderMargin();
+      if( dataCell.m_cellType == CellType::Separator )
+      {
+        size_t separatorLength = dataCell.m_cellWidth;
+        separatorLength += (idxColumn == numColumns - 1) ?
+                           tableLayout.getBorderMargin() * 2 + 2 :
+                           tableLayout.getColumnMargin();
+
+        dataCell.m_lines[0] = std::string( separatorLength, '-' );
+      }
+    }
 
-    std::vector< string > maxStringPerColumn;
-    for( auto const & column : columns )
+    // last pass for updating cells header
+    for( size_t rowIdx = 0; rowIdx < cellsHeaderLayout.size(); ++rowIdx )
     {
-      maxStringPerColumn.push_back( string( column.m_maxStringSize.length(), m_horizontalLine ) );
+      TableLayout::CellLayout & headerCell = cellsHeaderLayout[rowIdx][idxColumn];
+      TableLayout::CellLayout * previousHeaderCell = (idxColumn > 0) ?
+                                                     &cellsHeaderLayout[rowIdx][idxColumn - 1]:
+                                                     nullptr;
+
+      updateCellMaxLength( headerCell, maxColumnSize, previousHeaderCell );
     }
+  }
+}
 
-    string const patternBetweenColumns = GEOS_FMT( "{:-^{}}", m_horizontalLine, columnMargin );
+void TableTextFormatter::calculateTableSeparators( TableLayout & tableLayout,
+                                                   CellLayoutRows & cellsHeaderLayout,
+                                                   CellLayoutRows & cellsDataLayout,
+                                                   string & separatorLine ) const
+{
```

Comment:
I would rename to `adjustTableWidth()`, to emphazis that this method affect the table width, and that it sub-calls `adjustColumnWidths()`.